server/sonar-server/src/main/java/org/sonar/server/computation/source/SymbolsLineReader.java

```diff
@@ -77,19 +77,23 @@ public void read(DbFileSources.Line.Builder lineBuilder) {
 
   private void processSymbols(DbFileSources.Line.Builder lineBuilder) {
     int line = lineBuilder.getLine();
-    List<BatchReport.Symbol> lineSymbols = findSymbolsMatchingLine(line);
+
+    List<BatchReport.Symbol> lineSymbols = new ArrayList<>(this.symbolsPerLine.get(line));
+    // Sort symbols to have deterministic results and avoid false variation that would lead to an unnecessary update of the source files
+    // data
+    Collections.sort(lineSymbols, SymbolsComparator.INSTANCE);
```

Comment:
If you replace the SetMultimap by a SortedSetMultimap, I think you could avoid this sort, or am I wrong ?